test/docker/spark/Dockerfile

```diff
@@ -1,8 +1,11 @@
 FROM sequenceiq/hadoop-ubuntu:2.6.0
 
+# We are using 'sync' to separate chmod and script execution. See:
+# https://github.com/moby/moby/issues/9547 for details
+
 # Get latest java:
 COPY bin/setup_java.sh /setup_java.sh
-RUN chmod +x /setup_java.sh ; /setup_java.sh
+RUN chmod +x /setup_java.sh; sync; /setup_java.sh;
```

Comment:
My understanding of https://github.com/moby/moby/issues/9547#issuecomment-337684598 is that the `sync` workaround may fail randomly.

I think we should get rid of `chmod` and simply run `sh /setup_java.sh` or `bash /setup_java.sh`.

Reply:
We can't get rid of a command which is necessary.

Reply:
Why is it necessary?

Reply:
because `sh setup_java.sh` will also not work without changing the mode (as I recall, I have already tried those options while writing the Dockerfile). 

> My understanding of moby/moby#9547 (comment) is that the sync workaround may fail randomly.

I am pretty positive it will work (for our use case). We can always get back on it if it causes us issues in future.

Reply:
Why do you think it will work (despite the explanation in https://github.com/moby/moby/issues/9547#issuecomment-337684598)?

Reply:
I suggest trying the workaround in https://github.com/moby/moby/issues/9547#issuecomment-337684598 . It's more likely to work, and the downside (that we create an extra layer) isn't a big deal for test systems.

If that doesn't work for some reason let's stick with `sync` since it's better than nothing. It's unfortunate how terrible Docker is but that's something we need to deal with.